test/api/model/model_test.dart

```diff
@@ -108,4 +110,26 @@ void main() {
         .deepEquals([2, 3, 11]);
     });
   });
+
+  group('RealmDefaultExternalAccount', () {
+    final Map<String, dynamic> baseJson = Map.unmodifiable({
+      'name': 'site',
+      'hint': 'hint for site',
+      'text': 'description for site',
+      'url_pattern': 'http://example/%{username}',
+    });
+
+    RealmDefaultExternalAccount mkRealmDefaultExternalAccount(Map<String, dynamic> specialJson) {
+      return RealmDefaultExternalAccount.fromJson({ ...baseJson, ...specialJson });
+    }
+
+    test('realm_default_external_account', () {
+      final realmDefaultExternalAccount = mkRealmDefaultExternalAccount({});
+      check(realmDefaultExternalAccount)
+        ..name.equals('site')
```

Comment:
It looks like this is basically testing the `fromJson` method, and so checking that e.g. the `name` field it gets equals the value of the `name` property in the JSON.

That means that when it says `equals('site')`, the fact that `'site'` is the same value as appears in the input JSON is an essential part of the test. Because it's essential to the test, it's best if that fact can be found within the test case's own source code. For example like this:
```suggestion
    test('realm_default_external_account', () {
      final realmDefaultExternalAccount = mkRealmDefaultExternalAccount({
        'name': 'site',
        'hint': 'hint for site',
        'text': 'description for site',
        'url_pattern': 'http://example/%{username}',
      });
      check(realmDefaultExternalAccount)
        ..name.equals('site')
```

Where the `baseJson` pattern comes in handy, on the other hand (as well as other ways of expressing default/baseline test data outside an individual test), is for all the data whose specifics are irrelevant to a given test, where the test just needs *something* there that makes the data well-formed.

---

So for example up at the top of this file:
```dart
    test('delivery_email', () {
      check(mkUser({'delivery_email': 'name@email.com'}).deliveryEmailStaleDoNotUse)
        .equals('name@email.com');
```
that test doesn't care about any of the other properties in the input JSON, except that they have to be there and have appropriate types so that `fromJson` doesn't throw. So the `baseJson` encoded in `mkUser` serves it very well for those. It only cares about `delivery_email` — and so it specifies that one property explicitly.

We could have written that test like this:
```dart
    test('delivery_email', () {
      check(mkUser({}).deliveryEmailStaleDoNotUse)
        .equals('name@example.com');
```
relying on the `delivery_email` that's supplied in `baseJson`. But then the test wouldn't be self-contained — for the reader to understand what the test is trying to say, they'd need to go consult what's in `baseJson`. So instead we make the intended `delivery_email` value explicit.

Reply:
For this particular test, it's really just testing the auto-generated code for `fromJson`, because we happily have no interesting customizations on this type. So what I'd therefore actually prefer to do is to leave the test out.

Where we do have customizations of our own, especially any nontrivial ones, it's good to test those specifically. That's what you see happening earlier in this file on e.g. `User` — we test the handling of `profileData` and `isSystemBot`, which involve code of our own. Also of `deliveryEmailStaleDoNotUse`, which is a bit more marginal but does have our own customization in the form of that `JsonKey` annotation.